core/chat/persistence.go

```diff
@@ -48,7 +51,10 @@ func addMessage(message models.ChatMessage) {
 	if err != nil {
 		log.Fatal(err)
 	}
-	tx.Commit()
+	err = tx.Commit()
+	if err != nil {
+		log.Fatal(err)
+	}
 
 	defer stmt.Close()
```

Comment:
I think this defer should be moved up. Granted, currently we'd always abort if an error happens between the `stmt` declaration and this defer, but this could change and would potentially leak resources.